spec/rack_spec.rb

```diff
@@ -0,0 +1,72 @@
+require_relative "spec_helper"
+require "rack/test"
+require "rescue_registry"
+
+RSpec.describe "Rack App Usage" do
+  include Rack::Test::Methods
+
+  class CustomError < StandardError; end
+
+  let :app do
+    Rack::Builder.new do
+      use Rack::CommonLogger
+      # Catch stuff not handled by RescueRegistry
+      use Rack::ShowExceptions
+      # Handle registered exceptions
+      use RescueRegistry::ShowExceptions
+
+      map "/" do
+        run lambda { |env| [200, {'Content-Type' => 'text/plain'}, ['OK']] }
+      end
+
+      map "/error" do
+        run lambda { |env| raise "Error" }
+      end
+
+      map "/registered-error" do
+        run lambda { |env| raise CustomError }
+      end
+    end
+  end
+
+  let :context do
+    Module.new do
+      include RescueRegistry::Context
+
+      register_exception CustomError, status: 403
+    end
+  end
+
+  before do
+    RescueRegistry.context = context
```

Comment:
We should probably cleanup the context when the spec is done.

Reply:
maybe `around do / RescueRegistry.with_context`